### ISSUE TITLE
[YSQL] COPY buffering causes unique constraint violation

### ISSUE THREAD

**Issue by d-uspenskiy** [yugabyte/yugabyte-db#3219]
YSQL `COPY` operation uses write operation buffering optimization. But it works improperly in scenario like this one:
```
yugabyte=# create table t(k int primary key, v int);
CREATE TABLE
yugabyte=# copy t from stdin;
Enter data to be copied followed by a newline.
End with a backslash and a period on a line by itself, or an EOF signal.
>> 1	1
>> 1	2
>> \.
COPY 2
yugabyte=# select * from t;
 k | v 
---+---
 1 | 2
(1 row)
```

Vanilla postgres respond with the following error in same scenario

```
postgres=# copy t from stdin;
Enter data to be copied followed by a newline.
End with a backslash and a period on a line by itself, or an EOF signal.
>> 1	1
>> 1	2
>> \.
ERROR:  duplicate key value violates unique constraint "t_pkey"
DETAIL:  Key (k)=(1) already exists.
CONTEXT:  COPY t, line 2 
```

The reason of this behavior is that second operation in batched RPC doesn't see the results of first one. 
Batched RPC request

```
I1223 10:47:59.931311  6399 tablet_service.cc:849] Received Write RPC: tablet_id: "9f46d53b34e9486e80b138bd4314093d"
propagated_hybrid_time: 6459749498598227968
include_trace: false
write_batch {
  transaction {
    transaction_id: "\305<\221\\\210\177C\004\221\027,\002\201\214_b"
    isolation: SNAPSHOT_ISOLATION
    status_tablet: "f16efef5dbfc4cf19fc97761e488f88a"
    priority: 12518080117709118713
    DEPRECATED_start_hybrid_time: 18446744073709551614
  }
  DEPRECATED_may_have_metadata: true
}
read_time {
  read_ht: 6459749498552950784
  local_limit_ht: 6459749498757750784
  global_limit_ht: 6459749498757750784
  in_txn_limit_ht: 18446744073709551615
}
pgsql_write_batch {
  client: YQL_CLIENT_PGSQL
  stmt_type: PGSQL_INSERT
  table_id: "000030a9000030008000000000004000"
  schema_version: 0
  hash_code: 25390
  ybctid_column_value {
    value {
      binary_value: "H\200\000\000\002!"
    }
  }
  column_values {
    column_id: 11
    expr {
      value {
        int32_value: 2
      }
    }
  }
  column_refs {
  }
  ysql_catalog_version: 1
}
pgsql_write_batch {
  client: YQL_CLIENT_PGSQL
  stmt_type: PGSQL_INSERT
  table_id: "000030a9000030008000000000004000"
  schema_version: 0
  hash_code: 17821
  ybctid_column_value {
    value {
      binary_value: "H\200\000\000\002!"
    }
  }
  column_values {
    column_id: 11
    expr {
      value {
        int32_value: 3
      }
    }
  }
  column_refs {
  }
  ysql_catalog_version: 1
}
```

**Comment by frozenspider** [comment]
Related issue: #3445

**Comment by d-uspenskiy** [comment]
Issue will be fixed in context of #2934
